src/main/java/gift/service/GiftService.java

```diff
@@ -2,48 +2,57 @@
 
 
 import gift.model.Gift;
-import gift.model.GiftDao;
 import gift.model.GiftRequest;
 import gift.model.GiftResponse;
+import gift.repository.GiftRepository;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 import java.util.stream.Collectors;
 
 @Service
 public class GiftService {
 
-    private final GiftDao giftDao;
+    private final GiftRepository giftRepository;
 
-    public GiftService(GiftDao giftDao) {
-        this.giftDao = giftDao;
+    @Autowired
+    public GiftService(GiftRepository giftRepository) {
+        this.giftRepository = giftRepository;
     }
 
     public List<GiftResponse> getAllGifts() {
-        List<Gift> gifts = giftDao.findAll();
+        List<Gift> gifts = giftRepository.findAll();
         return gifts.stream()
                 .map(GiftResponse::from)
                 .collect(Collectors.toList());
     }
 
     public GiftResponse getGift(Long id) {
-        Gift gift = giftDao.findById(id);
+        Gift gift = giftRepository.findById(id)
+                .orElseThrow(() -> new NoSuchElementException("Gift not found with id " + id));
         return GiftResponse.from(gift);
     }
 
 
     public void addGift(GiftRequest giftRequest) {
         Gift gift = giftRequest.toEntity();
-        giftDao.create(gift);
+        giftRepository.save(gift);
     }
 
+    @Transactional
     public void updateGift(GiftRequest giftReq,Long id) {
-        Gift gift = giftReq.toEntity(id);
-        giftDao.updateById(gift,id);
+        Gift gift = giftRepository.findById(id)
+                .orElseThrow(() -> new NoSuchElementException("Gift not found with id " + id));
+        gift.modifyGift(giftReq.getName(),giftReq.getPrice(),giftReq.getImageUrl());
```

Comment:
gift라는 객체에 대해 수정을 명령하고 있으므로, 줄여도 될 것 같아요ㅎㅎ
```suggestion
        gift.modify(giftReq.getName(),giftReq.getPrice(),giftReq.getImageUrl());
```

Reply:
이미 gift 객체에 대한 메소드라 뒤에 굳이 명사를 안붙혀도 되는거였네요. 감사합니다.